docs/source/tracking/autolog.rst

```diff
@@ -0,0 +1,353 @@
+:py:func:`mlflow.autolog` before your training code. 
+
+.. code-block:: python
+
+    import mlflow
+
+    mlflow.autolog()
+
+    with mlflow.start_run():
+        # your training code goes here
+        ...
+
+This will enable MLflow to automatically log various information about your run, including:
+
+* **Metrics** - MLflow pre-selects a set of metrics to log, based on what model and library you use
+* **Parameters** - hyper params specified for the training, plus default values provided by the library if not explicitly set
+* **Model Signature** - logs :ref:`Model signature <model-signature>` instance, which describes input and output schema of the model
+* **Artifacts** -  e.g. model checkpoints
+* **Dataset** - dataset object used for training (if applicable), such as `tensorflow.data.Dataset`
+
+
+How to Get started
+==================
+
+Step 1 - Get MLflow
+-------------------
+
+MLflow is available on PyPI. If you don't already have it installed on your system, you can install it with:
+
+.. code-section::
+
+    .. code-block:: bash
+        :name: download-mlflow
+
+        pip install mlflow
+
+Step 2 - Insert ``mlflow.autolog`` in Your Code
+-----------------------------------------------
+
+For example, following code snippet shows how to enable autologging for a scikit-learn model:
+
+.. code-block:: python
+
+    import mlflow
+
+    from sklearn.model_selection import train_test_split
+    from sklearn.datasets import load_diabetes
+    from sklearn.ensemble import RandomForestRegressor
+
+    mlflow.autolog()
+
+    db = load_diabetes()
+    X_train, X_test, y_train, y_test = train_test_split(db.data, db.target)
+
+    rf = RandomForestRegressor(n_estimators=100, max_depth=6, max_features=3)
+    rf.fit(X_train, y_train)
+
+Step 3 - Execute Your Code
+--------------------------
+
+.. code-section::
+
+    .. code-block:: bash
+        :name: execute code
+
+        python YOUR_ML_CODE.py
+
+
+Step 4 - View Your Results in the MLflow UI
+-------------------------------------------
+
+Once your training job finishes, you can run following command to launch the MLflow UI:
+
+.. code-section::
+
+    .. code-block:: bash
+        :name: view-results
+
+        mlflow ui --port 8080
+
+Then, navigate to `http://localhost:8080 <http://localhost:8080>`_ in your browser to view the results.
+
+Customize Autologging Behavior
+==============================
+
+You can also control the behavior of autologging by passing arguments to :py:func:`mlflow.autolog` function.
+For example, you can disable logging of model checkpoints and assosiate tags with your run as follows:
+
+.. code-block:: python
+
+    import mlflow
+
+    mlflow.autolog(
+        log_model_signatures=False,
+        extra_tags={"YOUR_TAG": "VALUE"},
+    )
+
+See :py:func:`mlflow.autolog` for the full set of arguments you can use.
+
```

Comment:
Can we add a section with an example for disabling a particular flavor's autolog functionality (we don't need a full example in code for why this would be needed, but we can mention that if someone is using sklearn for building a pre-processing feature engineering step, they might not want that model to be automatically logged; but rather, they want the torch model that consumes that data to be autologged)

Reply:
Ah that is an interesting but common use case, will add that example, thanks!